src/Evolution/DiscontinuousGalerkin/Actions/ApplyBoundaryCorrections.hpp

```diff
@@ -302,6 +489,23 @@ bool receive_boundary_data_local_time_stepping(
       },
       box, db::get<::domain::Tags::Element<Dim>>(*box));
 
+  if (not have_all_intermediate_messages) {
+    return false;
+  }
+
+  if constexpr (std::is_same_v<
+                    evolution::dg::AtomicInboxBoundaryData<Dim>,
+                    typename evolution::dg::Tags::
+                        BoundaryCorrectionAndGhostCellsInbox<Dim>::type>) {
+    // We only decrease the counter if we are done with the current time
+    // and we only decrease it by the number of neighbors at the current
+    // time.
+    tuples::get<evolution::dg::Tags::BoundaryCorrectionAndGhostCellsInbox<Dim>>(
+        *inboxes)
+        .message_count.fetch_sub(
+            db::get<domain::Tags::Element<Dim>>(*box).number_of_neighbors(),
+            std::memory_order_acq_rel);
```

Comment:
It looks like this subtraction happens whenever `ApplyBoundaryCorrections::is_ready` returns true in LTS mode, even if no new messages were actually processed.  This happens for every dense-output operation.  Based on the field name, it looks like it's also assuming you get one message from each neighbor, which doesn't sound right if they're stepping at different rates.

If this should be the number of DG boundary messages processed, it's probably easiest to just add a counter in the loop above.

Reply:
Good point. This is code I never got fully working yet and is related to what we'd discussed a couple weeks ago.

Essentially, I think this should subtract the number of neighbors since the neighbors should only increment this when they send data for the next (complete) time step (I need to think about this more based on our previous convo). I don't think this should ever decrement on dense output, only on an actual time step. Is wrapping it in a `not DenseOutput` enough for now? It'll require quite a bit more testing in real use cases and decisions about the algorithm. I can also add an ERROR if you want

Reply:
What do you do when you don't need any messages from a neighbor?

It sounds like this case still needs a lot of design work.  Making it ERROR for now sounds good.

Reply:
Sounds good! Yes, the LTS stuff isn't working yet. I'm writing tests for the threaded callbacks and once that's done I'll be focusing on LTS support.